relay-crash/src/lib.rs

```diff
@@ -13,19 +14,63 @@ mod native {
     include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 }
 
+/// A transport function used to send envelopes to Sentry.
+pub type Transport = fn(envelope: &[u8]);
+
+/// Serializes a sentry_native envelope and passes the buffer to the [`Transport`] function.
+unsafe extern "C" fn transport_proxy(
+    envelope: *const native::sentry_envelope_s,
+    tx_pointer: *mut std::ffi::c_void,
+) {
+    if envelope.is_null() || tx_pointer.is_null() {
```

Comment:
This is overly defensive, since the Native SDK will not send a null pointer for the envelope, and the transport pointer is set below.